Terminal.Gui/ConsoleDrivers/AnsiEscapeSequence/AnsiEscapeSequenceRequest.cs

```diff
@@ -0,0 +1,197 @@
+﻿#nullable enable
+namespace Terminal.Gui;
+
+/// <summary>
+///     Describes an ongoing ANSI request sent to the console.
+///     Use <see cref="ResponseReceived"/> to handle the response
+///     when console answers the request.
+/// </summary>
+public class AnsiEscapeSequenceRequest
+{
+    /// <summary>
+    ///     Request to send e.g. see
+    ///     <see>
+    ///         <cref>EscSeqUtils.CSI_SendDeviceAttributes.Request</cref>
+    ///     </see>
+    /// </summary>
+    public required string Request { get; init; }
+
+    /// <summary>
+    ///     Invoked when the console responds with an ANSI response code that matches the
+    ///     <see cref="Terminator"/>
+    /// </summary>
+    public event EventHandler<AnsiEscapeSequenceResponse>? ResponseReceived;
+
+    /// <summary>
+    ///     <para>
+    ///         The terminator that uniquely identifies the type of response as responded
+    ///         by the console. e.g. for
+    ///         <see>
+    ///             <cref>EscSeqUtils.CSI_SendDeviceAttributes.Request</cref>
+    ///         </see>
+    ///         the terminator is
+    ///         <see>
+    ///             <cref>EscSeqUtils.CSI_SendDeviceAttributes.Terminator</cref>
+    ///         </see>
+    ///         .
+    ///     </para>
+    ///     <para>
+    ///         After sending a request, the first response with matching terminator will be matched
+    ///         to the oldest outstanding request.
+    ///     </para>
+    /// </summary>
+    public required string Terminator { get; init; }
+
+    /// <summary>
+    ///     Execute an ANSI escape sequence escape which may return a response or error.
+    /// </summary>
+    /// <param name="ansiRequest">The ANSI escape sequence to request.</param>
+    /// <param name="result">
+    ///     When this method returns <see langword="true"/>, an object containing the response with an empty
+    ///     error.
+    /// </param>
+    /// <returns>A <see cref="AnsiEscapeSequenceResponse"/> with the response, error, terminator and value.</returns>
+    public static bool TryExecuteAnsiRequest (AnsiEscapeSequenceRequest ansiRequest, out AnsiEscapeSequenceResponse result)
+    {
+        var response = new StringBuilder ();
+        var error = new StringBuilder ();
+        var savedIsReportingMouseMoves = false;
+        NetDriver? netDriver = null;
+        var values = new string? [] { null };
+
+        try
+        {
+            switch (Application.Driver)
+            {
+                case NetDriver:
```

Comment:
Please refactor this such that this driver-dependent code is in the related driver. This is really poor encpasulation.

Reply:
Done.